cmd/sensors/src/lib.rs

```diff
@@ -52,6 +53,12 @@ struct SensorsArgs {
     #[clap(long, short)]
     list: bool,
 
+    /// use the given backend to read sensor data
+    ///
+    /// [default: `hiffy` if present, otherwise `readmem`]
+    #[clap(long, short, value_enum)]
+    backend: Option<Backend>,
```

Comment:
i'm not sure if i get the default behavior around this: do you mean that if you pass `--backend` without an argument, it uses `hiffy`, and otherwise, it uses RAM reading? or just, it uses RAM-reading by default?

because, we could express the latter by writing
```suggestion
    #[clap(long, short, value_enum, default_value_t=Backend::Readmem)]
    backend: Backend,
```
avoiding the need to make this an `Option`?

Reply:
The comment here was unclear ("if present" meant "if hiffy was present on the target", not "if the `--backend` flag is present in the CLI").

The intention was to normally use hiffy with an automatic fallback to RAM reading (making the default case robust), but you could override it if you want to force one or the other (hence the `Option`).

Let me know if 019575ef3dee0c0b9787019f0ce18bf5beb8648d is more obvious

Reply:
yeah, i think that makes substantially more sense. thanks!